op-node/rollup/derive/batches.go

```diff
@@ -34,12 +35,15 @@ func FilterBatches(log log.Logger, config *rollup.Config, epoch rollup.Epoch, mi
 	return
 }
 
-func ValidBatch(batch *BatchData, config *rollup.Config, epoch rollup.Epoch, minL2Time uint64, maxL2Time uint64) error {
+func ValidBatch(batch *BatchData, config *rollup.Config, epoch rollup.Epoch, epochHash common.Hash, minL2Time uint64, maxL2Time uint64) error {
 	if batch.Epoch != epoch {
 		// Batch was tagged for past or future epoch,
 		// i.e. it was included too late or depends on the given L1 block to be processed first.
 		return DifferentEpoch
 	}
+	if batch.OriginHash != epochHash {
```

Comment:
Can merge this with above check, and return a wrapped error around `DifferentEpoch` with the two different id values (batch vs epoch).

Reply:
check merged, but no differentiation in the error case